### PR TITLE
Checks that custom fields exist before saving 

### PR DESCRIPTION
**Pull request by spencerrlongg**
# Description
This adds a trait that checks that custom fields exist (and exist on the correct fieldset) before saving. 

This **cannot** be merged in it's current form until #14458 is merged, so draft for now. 

image

Fixes [sc-23936]

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
Manually tested for now, custom fields are difficult to test at the moment (Marcus can speak to that better than I can)

**Test Configuration**:
* PHP version: 8.1
* MySQL version 8.2

### CODE REVIEW

**Comment by spencerrlongg** [comment]
Was just curious and wanted to double check, this does actually merge with the error bag from the request's validator and it's the same validator instance, so all validation is included in the same response, which is pretty nice. 🙂

image

**Comment by snipe** [comment]
Quick followup here - anything we can do to get this out of WIP?

**Comment by uberbrady** [comment]
@spencerrlongg Can you re-target this to develop - maybe once you're a little further on out of WIP stage? Thank you!

**Comment by spencerrlongg** [comment]
> Quick followup here - anything we can do to get this out of WIP?

We can talk about #14458, the PR that's holding this one in WIP. 

> Can you re-target this to develop - maybe once you're a little further on out of WIP stage? Thank you!

It's retargeted, it's only WIP to make sure it doesn't get merged without #14458 - other than that it's ready to go.

**Comment by spencerrlongg** [comment]
This got lost somehow and left in drafts, but it should actually be good to go.

File: app/Http/Requests/Traits/MayContainCustomFields.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+{
+    // this gets called automatically on a form request
+    public function withValidator($validator)
+    {
+        // find the model
+        if ($this->method() == 'POST') {
+            $asset_model = AssetModel::find($this->model_id);
+        }
+        if ($this->method() == 'PATCH' || $this->method() == 'PUT') {
+            // this is dependent on the asset update request PR
```
Comment (snipe) [inline app/Http/Requests/Traits/MayContainCustomFields.php:18]: Was the dependent PR merged?

Comment (spencerrlongg) [reply app/Http/Requests/Traits/MayContainCustomFields.php:18]: Yep! #14458 6 days ago.

**Comment by snipe** [comment]
Nice work!

**Pull request merged**